opba-protocols/sandboxes/hbci-sandbox/src/main/java/de/adorsys/opba/protocol/sandbox/hbci/domain/HbciSandboxPayment.java

```diff
@@ -93,7 +93,7 @@ public String getModifiedAtDateString() {
 
     public Transaction toTransaction(String accountNumber, BigDecimal balance) {
         Transaction transaction = new Transaction();
-        transaction.setAmount(getAmount().toString());
+        transaction.setAmount(getAmount().negate().toString());
```

Comment:
Correct amount sign should be handled in:
```
if (getDeduceFrom().endsWith(accountNumber)) {
            transaction.setBalanceAfter(balance.subtract(getAmount()).toString());
        } else {
            transaction.setBalanceAfter(balance.add(getAmount()).toString());
        }
```
because the transaction amount sign depends on whether is this account `from` or `to`

Reply:
moved set amount inside if